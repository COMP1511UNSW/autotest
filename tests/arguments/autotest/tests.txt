files=echo.sh
program=./echo.sh

1 expected_stdout="\n"
2 arguments=hello expected_stdout="hello\n"
3 arguments=hello world  expected_stdout="hello world\n"
arguments=["hello","world"]
4  expected_stdout="hello world\n"