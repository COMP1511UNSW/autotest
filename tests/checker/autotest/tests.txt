program=hello.sh

0 expected_stdout="hello\n"

checkers=checker.sh
pre_compile_command="pre_compile.sh *"

1 expected_stdout="hello\n"
