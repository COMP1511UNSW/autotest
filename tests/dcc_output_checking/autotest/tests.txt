files="hello.c"
dcc_output_checking=true
expected_stdout="hello\nworld\n"
ignore_case=1
postprocess_output_command="sed 's/0x[a-z0-9]*/0x/g'"

implicit_output_checking_pass arguments=hello world 
implicit_output_checking_fail arguments=good-bye world 


dcc_output_checking=false
explicit_no_output_checking_pass arguments=hello World 
explicit_no_output_checking_fail arguments=good-bye World 

dcc_output_checking=true
explicit_output_checking_pass arguments=Hello world 
explicit_output_checking_fail arguments=Good-bye world 
