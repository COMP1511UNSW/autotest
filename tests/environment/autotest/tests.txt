files="a.sh"
environment_set={'answer' :42 }

set_environment_variable command="echo $answer"
set_environment_variable expected_stdout="42\n"

# SAMPLE_ENVIRONMENT_VARIABLE set in do_tests.sh
filtered_environment_variable command="echo $SAMPLE_ENVIRONMENT_VARIABLE"
filtered_environment_variable expected_stdout="\n"

environment_kept='SAMPLE_ENVIRONMENT_VARIABLE'
environment_kept command="echo $SAMPLE_ENVIRONMENT_VARIABLE"
environment_kept expected_stdout="sample_value\n"
