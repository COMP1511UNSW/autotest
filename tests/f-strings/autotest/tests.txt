program=echo.sh
_var=hello

1 arguments=f'{_var} {6 * 7} {program}' expected_stdout="hello 42 echo.sh\n"