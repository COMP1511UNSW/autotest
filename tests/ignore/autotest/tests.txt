program=echo.sh
expected_stdout="Hello World!\n"

ignore_case ignore_case=1 arguments="hello WORLD!"

ignore_whitespace ignore_whitespace=1 arguments=" Hello\tWorld  !"

ignore_trailing_whitespace ignore_trailing_whitespace=1 arguments="Hello World!    "

ignore_characters ignore_characters="!@" arguments="@Hello World@"

compare_only_characters compare_only_characters="eW" arguments="ello Wrld@"

ignore_blank_lines ignore_blank_lines=True arguments="\nHello World!\n\n"

postprocess_output_command postprocess_output_command="sed s/z//g"  arguments="zHello zWorld!z" 
