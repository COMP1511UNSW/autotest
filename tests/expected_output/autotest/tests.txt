program=echo.sh
arguments="Hello World!"

expected_stdout expected_stdout="Hello World!\n"
#expected_stdout_file expected_stdout=['hello.txt']
#expected_stdout_file_deprocated expected_stdout_file='hello.txt'

#command="echo 1 >one.txt; echo 2 >two.txt; echo 3 >three.txt; echo stdout; echo stderr 1>&2"
#expected_output_pass expected_stdout='stdout\n' expected_stderr='stderr\n' expected_files={"one.txt":"1\n", "two.txt":"2\n","three.txt":"3\n"}
#expected_output_fail expected_stdout='stdout\n' expected_stderr='stderr\n' expected_files={"one.txt":"1\n", "two.txt":"42\n","three.txt":"3\n"}
#
#stdin stdin="Hello World!\n" command="cat" expected_stdout="Hello World!\n"
#stdin_file stdin=['hello.txt'] command="cat" expected_stdout="Hello World!\n"
#stdin_file_deprocated stdin_file=hello.txt command="cat" expected_stdout="Hello World!\n"
#
command="echo 1 >one.txt"
expected_file_deprocated_1 expected_file_name="one.txt"
